p2p/client.go

```diff
@@ -313,6 +337,36 @@ func (c *Client) setupPeerDiscovery(ctx context.Context) error {
 	return nil
 }
 
+func (c *Client) setupBlockedPeers(peers []peer.AddrInfo) error {
+	if c.gater == nil {
+		return fmt.Errorf("connection gater is nil, cannot blacklist peers")
+	}
+	if len(peers) > 0 {
+		for _, p := range peers {
+			err := c.gater.BlockPeer(p.ID)
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
`range` handle empty slices and `nil`s perfectly fine. You can simplify the code.

```suggestion
	for _, p := range peers {
		err := c.gater.BlockPeer(p.ID)
		if err != nil {
			return err
		}
	}
```